### PR TITLE
Provide configure script and installation description for HDF5

### LINKED ISSUE #410: HDF5 is enabled by default, but no installation description is provided
The default setting in `config_dealii.sh` is `DEAL_II_WITH_HDF5:BOOL="ON"`, but the wiki page https://github.com/exadg/exadg/wiki/Installation does not provide an installation description for HDF5.

### PR DESCRIPTION
**Pull request by nfehn**
closes #410 

Together with the configure script introduced here, let us discuss the installation description for HDF5 in the wiki https://github.com/exadg/exadg/wiki/Installation during this PR. @jh66637 I did not understand whether `sudo apt install ...` and the variant with `.configure ...` are two different alternatives or not. In particular, I did not know what exactly you mean by `download hdf5`. Could you explain this?

### CODE REVIEW

File: scripts/config_hdf5.sh [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+#########################################################################
+
+
+./configure --prefix=$WORKING_DIRECTORY/hdf5/install/ --enable-unsupported --enable-symbols=yes --disable-threadsafe --disable-cxx --enable-hl --without-szlib --enable-parallel --enable-shared CC=mpicc
```
Comment (nfehn) [inline scripts/config_hdf5.sh:30]: I think we should use separate rows for every option for readability, agree? See for comparison e.g. the configure script for PETSc.

Comment (jh66637) [reply scripts/config_hdf5.sh:30]: I agree, I just copied the line from a script I had.

**Comment by jh66637** [comment]
Download HDF5 would be 
```
wget https://github.com/HDFGroup/hdf5/archive/refs/tags/hdf5-1_12_2.tar.gz
tar -xzf  hdf5-1_12_2.tar.gz
cd hdf5-hdf5-1_12_2/
```

**Comment by jh66637** [comment]
I think the wiki entry looks good. Just some minor comments:
1) "Next, go to your working directory" -> "If you want to install HDF5 from sources"
2) ```make``` -> ```make -j<N>```
3) I am pretty sure that one has to provide the HDF5 library directory and include directory in the deal.ii configure script if installation is done from sources. Since I am running on a system with environment modules, this is not a problem since ```module load``` exports the corresponding path. I am currently not able to check it without environment modules, though.

**Comment by nfehn** [comment]
So the manual installation is an alternative to `sudo apt ...`. Or do I still have to run the configure-script/make command?

**Comment by nfehn** [comment]
could you check again the wiki page?

**Comment by jh66637** [comment]
> So the manual installation is an alternative to `sudo apt ...`. Or do I still have to run the configure-script/make command?

It's an alternative on systems where you can not use a package manager.

**Comment by jh66637** [comment]
Regarding the Wiki:
If you could change the HDF5 section to HDF5 (optional) I think it is more consistent to other optional packages. Before doing ```wget``` I would probably ```cd hdf5```, but both should work; it's probably a matter of taste.

**Comment by nfehn** [comment]
I made these changes as suggested

**Comment by jh66637** [comment]
It would be best if someone else can confirm the installation procedure works. In general, I am confident since it worked for me some month ago. So this is OK from my side once the line breaks in the configure script are added :)

**Review by jh66637 — APPROVED** [review]

**Pull request merged**